### PR TITLE
Added setListCountAndRef and setStrListCountAndRef.

### PR DESCRIPTION
**Pull request by Rotaerk**
These allow you to set both an array-pointer field and its corresponding count field simultaneously from a list, which is a common thing to do in Vulkan.
Also changed setStrRef to set the pointer to null in the case of an empty string.

### CODE REVIEW

**Comment by Rotaerk** [comment]
I'm not certain if you're *always* going to need to set both a pointer and count together, but so far everywhere I've used `setListRef` or `setStrListRef`, I've also had to set a corresponding count field.  These helpers turn code that looks like this:
```hs
set @"enabledLayerCount" (fromIntegral $ length validationLayers) &*
setStrListRef @"ppEnabledLayerNames" validationLayers &*
set @"enabledExtensionCount" (fromIntegral $ length extensions) &*
setListRef @"ppEnabledExtensionNames" extensions
```

into this:
```hs
setStrListCountAndRef @"enabledLayerCount" @"ppEnabledLayerNames" validationLayers &*
setListCountAndRef @"enabledExtensionCount" @"ppEnabledExtensionNames" extensions
```

**Comment by Rotaerk** [comment]
I should also note:  These can only be implemented here, not as helper functions in calling code, because they rely on the Union type family that Create.hs doesn't export.

**Comment by achirkin** [comment]
Ok, why not :)  Thanks!
Probably, I should better expose the internals at some point.

File: vulkan-api/src/Graphics/Vulkan/Marshal/Create.hs [hunk 1]
```diff
@@ -170,11 +171,15 @@ setStr v = CreateVkStruct $ \p ->
...
 --   This function also attaches a reliable finalizer to the vulkan struct,
 --    so that the allocated memory is freed when the structure is GCed.
+--
+--   This function writes null pointer if used with an empty string.
 setStrRef :: forall fname x
            . ( CanWriteField fname x
              , FieldType fname x ~ CString
              )
           => String -> CreateVkStruct x '[fname] ()
+setStrRef "" = CreateVkStruct $ \p ->
```
Comment (achirkin) [inline vulkan-api/src/Graphics/Vulkan/Marshal/Create.hs:181]: Are you sure nobody ever needs to set an empty string rather than NULL?

**Pull request merged**